src/components/dialogs/dataSources/COVIDcast.svelte

```diff
@@ -23,34 +24,41 @@
     }
   }
 
-  onMount(() => {
-    fetchCOVIDcastMeta().then((res) => {
-      geoTypes = [...new Set(res.map((d) => d.geo_type))];
-      const byDataSource = new Map<string, LabelValue & { signals: string[] }>();
-      for (const row of res) {
-        const ds = byDataSource.get(row.data_source);
-        if (!ds) {
-          byDataSource.set(row.data_source, {
-            label: row.data_source,
-            value: row.data_source,
-            signals: [row.signal],
-          });
-        } else if (!ds.signals.includes(row.signal)) {
-          ds.signals.push(row.signal);
+  function fetchMetadata() {
+    fetchCOVIDcastMeta((api_key = api_key)).then((res) => {
+      if (res.length !== 0) {
```

Comment:
Something to consider before the PR is merged: what if the API key is invalid?

This can be determined before the call to the `covidcast` endpoint; if the API key is incorrect, the `covidcast_meta` endpoint will return an empty result set. But how should this be communicated to the user?

Reply:
It is not just that it returns an empty set on an invalid key; it returns HTTP status code `401 Unauthorized`.  In this case, you should color the input box light red, add a tooltip that says "API Key invalid", but otherwise proceed as if the user specified no key at all (allow form submission but use no key in the request(s) to the API server).